### PR TITLE
 Make sslheaders plugin better conform to documentation.

### PR DESCRIPTION
**Pull request by ywkaras**
This PR is dependent on https://github.com/apache/trafficserver/pull/4700 .

### CODE REVIEW

File: plugins/experimental/sslheaders/sslheaders.cc [hunk 1]
```diff
@@ -116,6 +116,49 @@ SslHdrSetHeader(TSMBuffer mbuf, TSMLoc mhdr, const std::string &name, BIO *value
...
+    if (_x509 == _nonNullInvalidValue()) {
+      _set();
+    }
+
+    return _x509;
+  }
+
+  ~WrapX509()
+  {
+    if (IsClient && (_x509 != _nonNullInvalidValue()) and (_x509 != nullptr)) {
```
Comment (bryancall) [inline plugins/experimental/sslheaders/sslheaders.cc]: Please change the "and" to &&

Comment (ywkaras) [reply plugins/experimental/sslheaders/sslheaders.cc]: oops

**Comment by bryancall** [comment]
@ywkaras Once it is approved it will be merged into master and will be on what will be the 9.0.x branch.  I will cherry pick it into the 8.0.x branch and we will need a new PR created for the 7.1.x branch.  Please made the PR for 7.1.x after this PR has been approved and merged to master.

File: plugins/experimental/sslheaders/sslheaders.cc [hunk 2]
```diff
@@ -124,38 +167,39 @@ SslHdrExpand(SSL *ssl, const SslHdrInstance::expansion_list &expansions, TSMBuff
...
         break;
       case SSL_HEADERS_SCOPE_SERVER:
-        x509 = SSL_get_certificate(ssl);
+        x509 = serverX509.get();
+        if (x509 == nullptr) {
+          continue;
+        }
         break;
       default:
         x509 = nullptr;
```
Comment (ywkaras) [inline plugins/experimental/sslheaders/sslheaders.cc]: I'm not handling this case correctly, need to fix.

**Comment by bryancall** [comment]
Cherry picked to 8.0.2 release

**Pull request merged**